### PR TITLE
Update text in algorithm sections

### PR DESCRIPTION
**Pull request by guidou**
This addresses issues #16, #22, #45,  #40, #41, #42, , #43 and  #44

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -37,7 +37,7 @@
...
-        The unique ID of the audio device through which output is being delivered, or empty string if it is using the user agent default. This ID should be one of the <a href="https://w3c.github.io/mediacapture-main/#widl-MediaDeviceInfo-deviceId"><code>MediaDeviceInfo.deviceId</code></a> values returned from <a href="https://w3c.github.io/mediacapture-main/#widl-MediaDevices-enumerateDevices-void-MediaDeviceInfoCallback-resultCallback"><code>MediaDevices.enumerateDevices()</code></a> [[!GETUSERMEDIA]], or one of the predefined IDs listed below.
+        The unique ID of the audio device through which output is being delivered, or empty string if it is using the user agent default. This ID should be equal to the <a href="https://w3c.github.io/mediacapture-main/#dom-mediadeviceinfo-deviceid"><code>deviceId</code></a> attribute of one of the <a href="https://w3c.github.io/mediacapture-main/#idl-def-mediadeviceinfo"><code>MediaDeviceInfo</code></a> values returned from <a href="https://w3c.github.io/mediacapture-main/#dom-mediadevices-enumeratedevices"><code>MediaDevices.enumerateDevices()</code></a> [[!GETUSERMEDIA]], or one of the predefined IDs listed below.
```
Comment (alvestrand) [inline index.html]: Nit: "or empty string" -> "or the empty string". You might want to turn on auto line breaks; this line is very long.

File: index.html [hunk 2]
```diff
@@ -46,17 +46,16 @@
...
         <li><p>Let <var>promise</var> be a new promise.</p></li>
-        <li><p>Run the following steps asynchronously:</p>
+        <li><p>Run the following substeps in parallel:</p>
           <ol>
-            <li><p>If <code>sinkId</code> does not match any audio output device identified by <a href="https://w3c.github.io/mediacapture-main/#dom-mediadevices-enumeratedevices"><code>enumerateDevices()</code></a>, reject <var>promise</var> with a new <a href="http://www.w3.org/TR/html5/infrastructure.html#domexception"><code>DOMException</code></a> whose name is <a href="http://www.w3.org/TR/html5/infrastructure.html#notfounderror"><code>NotFoundError</code></a>.</p></li>
+            <li><p>If <code>sinkId</code> does not match any audio output device identified by the result the would be provided by <a href="https://w3c.github.io/mediacapture-main/#dom-mediadevices-enumeratedevices"><code>enumerateDevices()</code></a>, reject <var>promise</var> with a new <a href="http://www.w3.org/TR/html5/infrastructure.html#domexception"><code>DOMException</code></a> whose name is <a href="http://www.w3.org/TR/html5/infrastructure.html#notfounderror"><code>NotFoundError</code></a>.</p></li>
```
Comment (alvestrand) [inline index.html]: Nit: "the result the would be provided" -> "the result that would be provided"

File: index.html [hunk 3]
```diff
@@ -46,17 +46,16 @@
...
             <li><p>If application is not authorized to play audio through the device identified by <code>sinkId</code>, reject <var>promise</var> with a new <a href="http://www.w3.org/TR/html5/infrastructure.html#domexception"><code>DOMException</code></a> whose name is <a href="http://www.w3.org/TR/html5/infrastructure.html#securityerror"><code>SecurityError</code></a>.</p></li>
```
Comment (alvestrand) [inline index.html]: Nit: "If application" -> "If the application"

File: index.html [hunk 4]
```diff
@@ -46,17 +46,16 @@
...
+            <li><p>Switch the underlying output device for the media element to the device identified by <code>sinkId</code>.
+            <p class="note">If this substep is successful and the media element's <a href="http://www.w3.org/TR/html5/embedded-content-0.html#dom-media-paused"><code>paused</code></a> attribute is false, audio will stop playing out of the device represented by the element's <a data-link-for="HTMLMediaElement"><code>sinkId</code></a> attribute and will start playing out of the device identified by <code>sinkId</code></p></li>
+            <li><p>If the preceding substep failed, reject <var>promise</var> with a new <a href="http://www.w3.org/TR/html5/infrastructure.html#domexception"><code>DOMException</code></a> whose name is <a href="http://www.w3.org/TR/html5/infrastructure.html#aborterror"><code>AbortError</code></a> and abort these substeps.</p></li>
+            <li><p>Update the internal state of the media element so that the value of its <a  data-link-for="HTMLMediaElement"><code>sinkId</code></a> attribute corresponds to <code>sinkId</code>.</p></li>
```
Comment (alvestrand) [inline index.html]: Super-nit: one space between <a and data-link-for. running tidy might help?

File: index.html [hunk 5]
```diff
@@ -67,20 +66,20 @@
         <h3>Algorithms</h3>
         <section id="algorithms-sink-unavailable">
           <h4>Sink no longer available</h4>
-          <p>The audio device associated with <a  data-link-for="HTMLMediaElement"><code>HTMLMediaElement.sinkId</code></a> may become unavailable, for example if it is unplugged.</p>
-          <p>When the audio device identified by the <a  data-link-for="HTMLMediaElement"><code>sinkId</code></a> attribute is no longer available, the user agent must take no action. For example, if the media element's <a href="http://www.w3.org/TR/html5/embedded-content-0.html#dom-media-paused"><code>paused</code></a> attribute is false when the device identified by the sinkId is no longer available, then playback will continue as normal. Of course, audio will not be rendered because the device to which the media element is attached is unavailable.</p>
+          <p>The audio device identified by a media element's <a  data-link-for="HTMLMediaElement"><code>sinkId</code></a> attribute may become unavailable, for example, if it is unplugged.</p>
```
Comment (alvestrand) [inline index.html]: I think omitting the comma is correct - "for example" is a part of the following subsentence, not an insertion.

File: index.html [hunk 6]
```diff
@@ -67,20 +66,20 @@
...
+          <p>When the audio device identified by the <a  data-link-for="HTMLMediaElement"><code>sinkId</code></a> attribute is no longer available, the user agent must take no action. For example, if the media element's <a href="http://www.w3.org/TR/html5/embedded-content-0.html#dom-media-paused"><code>paused</code></a> attribute is false when the device identified by the sinkId is no longer available, then playback will continue as normal. In this case, audio will not be rendered because the device to which the media element is attached is unavailable.</p>
```
Comment (alvestrand) [inline index.html]: In this case,  the comma after "For example" is correct (I think. English is slippery!)

File: index.html [hunk 7]
```diff
@@ -147,7 +146,7 @@
...
+        <p>However, implementations must support implicit consent via the <a href="https://w3c.github.io/mediacapture-main/#dom-mediadevices-getusermedia"><code>getUserMedia()</code></a> permission prompt; when an audio input device is authorized and opened via <a href="https://w3c.github.io/mediacapture-main/#dom-mediadevices-getusermedia"><code>getUserMedia()</code></a>, this also authorizes access to any associated audio output devices (i.e., those with the same <a href="https://w3c.github.io/mediacapture-main/#dom-mediadeviceinfo-groupid"><code>groupId</code></a>). This conviently handles the common case of wanting to route both input and output audio through a headset or speakerphone device.</p>
```
Comment (alvestrand) [inline index.html]: suggest using "MUST support".

nit: conviently -> conveniently

File: index.html (hunk 2 above)
Comment (guidou) [inline index.html]: done

File: index.html (hunk 1 above)
Comment (guidou) [inline index.html]: done

File: index.html (hunk 3 above)
Comment (guidou) [inline index.html]: done

File: index.html (hunk 4 above)
Comment (guidou) [inline index.html]: Done. It was in a lot of places. Will apply tidy in an follow-up PR.

File: index.html (hunk 5 above)
Comment (guidou) [inline index.html]: Done

File: index.html (hunk 7 above)
Comment (guidou) [inline index.html]: Done

**Pull request merged**